app/battle.py

```diff
@@ -0,0 +1,16 @@
+from typing import Dict
+
+
+def battle(knight1: Dict, knight2: Dict) -> Dict[str, int]:
+    knight1["hp"] -= knight2["power"] - knight1["protection"]
+    knight2["hp"] -= knight1["power"] - knight2["protection"]
```

Comment:
The calculation for reducing `hp` does not account for the possibility that `protection` might be greater than `power`, which could result in increasing `hp`. Consider using `max(0, knight2['power'] - knight1['protection'])` to ensure `hp` is only reduced.